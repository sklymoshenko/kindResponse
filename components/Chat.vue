<template>
  <div class="rounded-md w-full md:w-1/2 my-2 border-secondary border">
    <BaseBubble :text="`Hi can you tell me why you are not at work ?`" />
    <BaseBubble :text="`I dont know`" :side="rightSide" />
  </div>
</template>

<script lang="ts">
import {
  Component,
  Vue
} from "nuxt-property-decorator";
import { Side } from "@/enums/enums.ts";

import BaseBubble from "@/components/Base/BaseBubble.vue";
@Component({ components: { BaseBubble } })
export default class Chat extends Vue {
  rightSide: Side = Side.RIGHT;
};
</script>

<style></style>
