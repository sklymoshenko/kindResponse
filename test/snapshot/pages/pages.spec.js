import { mount, createLocalVue } from "@vue/test-utils";

import BlurClosing from "@/directives/blurClosing.ts";
import Index from "@/pages/index.vue";
import Chat from "~/components/Chat.vue";

import { mockFetchedCategories } from "@/test/mocks.ts";
const { fetchedOptions } = mockFetchedCategories();

const localVue = createLocalVue();
localVue.directive("blur-closing", BlurClosing);

describe("Module tests", () => {
  describe("Index", () => {
    let wrapper;
    beforeEach(async() => {
      fetch.mockResponse(JSON.stringify(fetchedOptions));
      wrapper = mount(Index);
      await wrapper.findComponent(Chat).setProps("question", "Test question");
    });
    test("Should create a a vue instance", () => {
      expect(wrapper.vm).toBeTruthy();
    });
    test("Renders correctly", () => {
      expect(wrapper.element).toMatchSnapshot();
    });
    test("Set 'to'", async() => {
      await wrapper.vm.handleToChange("Mike");
      expect(wrapper.element).toMatchSnapshot();
    });
    test("Set 'from'", async() => {
      await wrapper.vm.handleFromChange("Boss");
      expect(wrapper.element).toMatchSnapshot();
    });
    test("Randon questions with different input ", async() => {
      await wrapper.vm.randomIntegerBetween(0, 1);
      expect(wrapper.element).toMatchSnapshot();
      await wrapper.vm.randomIntegerBetween(3, 1);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
