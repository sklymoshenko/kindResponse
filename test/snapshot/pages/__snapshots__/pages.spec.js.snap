// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module tests Index Renders correctly 1`] = `
<div
  class="container h-screen"
>
  <span
    class="flex flex-col justify-center items-center mt-12"
  >
    <span
      class="flex items-end"
    >
      <p
        class="title-test font-titles font-bold capitalize sm:text-lg md:text-2xl mr-1"
      >
        
      Kind responses for everybody
    
      </p>
       
      <svg
        class="icon-test stroke-current text-danger w-8"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </span>
   
  <div
    class="started-test flex flex-col md:flex-row mx-auto border-secondary border-solid border-2 w-4/5 z-10 origin-center"
  >
    <div
      class="wrapper w-full md:w-1/2 flex flex-col  mt-3 md:my-auto"
    >
      <div
        class="rounded-md w-full my-2 relative"
      >
        <label
          class="block text-sm font-medium text-gray-700 text-left pl-2"
          id="listbox-label"
        >
          
    From
  
        </label>
         
        <input
          class="focus:outline-none focus:border-secondary transition-colors duration-500 block w-full sm:text-lg rounded-sm font-primary border-b border-solid border-primary pl-2"
          id="From"
          label="From"
          options="[object Object]"
          placeholder="Mike"
          type="text"
        >
          <option
            disabled="disabled"
            selected="selected"
            value=""
          >
            
      From
    
          </option>
        </input>
      </div>
       
      <div
        class="rounded-md w-full my-2 relative"
      >
        <label
          class="block text-sm font-medium text-gray-700 text-left pl-2"
          id="listbox-label"
        >
          
    To
  
        </label>
         
        <input
          class="focus:outline-none focus:border-secondary transition-colors duration-500 block w-full sm:text-lg rounded-sm font-primary border-b border-solid border-primary pl-2"
          id="To"
          label="To"
          options="[object Object]"
          placeholder="My lovely Boss"
          type="text"
        >
          <option
            disabled="disabled"
            selected="selected"
            value=""
          >
            
      To
    
          </option>
        </input>
      </div>
       
      <div
        class="rounded-md w-full my-2 relative"
      >
        <label
          class="block text-sm font-medium text-gray-700 text-left pl-2"
          id="listbox-label"
        >
          
     Category
  
        </label>
         
        <div
          class="focus:outline-none focus:border-secondary transition-colors duration-500 block w-full sm:text-lg rounded-sm font-primary "
          id=" Category"
          label=" Category"
          placeholder=""
          type="select"
        >
          <button
            aria-expanded="true"
            aria-haspopup="listbox"
            aria-labelledby="listbox-label"
            class="relative w-full border border-primary rounded-md pl-3 pr-10 py-2 text-left cursor-default focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm excluded"
            id="dropdown-button"
            type="button"
          >
            <span
              class="flex items-center"
            >
              <span
                class="ml-3 block truncate"
              >
                
        test
      
              </span>
            </span>
             
            <span
              class="ml-3 absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
            >
              <svg
                aria-hidden="true"
                class="h-5 w-5 text-gray-400"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 3a1 1 0 01.707.293l3 3a1 1 0 01-1.414 1.414L10 5.414 7.707 7.707a1 1 0 01-1.414-1.414l3-3A1 1 0 0110 3zm-3.707 9.293a1 1 0 011.414 0L10 14.586l2.293-2.293a1 1 0 011.414 1.414l-3 3a1 1 0 01-1.414 0l-3-3a1 1 0 010-1.414z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
           
          <div
            class="absolute mt-1 w-full rounded-md bg-white shadow-lg z-10 hidden"
          >
            <ul
              aria-activedescendant="listbox-item-3"
              aria-labelledby="listbox-label"
              class="max-h-56 rounded-md py-1 text-base ring-1 ring-black ring-opacity-5 overflow-auto focus:outline-none sm:text-sm h-48"
              role="listbox"
              tabindex="-1"
            >
              <li
                class="text-gray-900 cursor-default select-none relative py-2 pl-3 pr-9 hover:bg-secondary hover:text-white"
                id="listbox-item-0"
                role="option"
              >
                <div
                  class="flex items-center"
                >
                  <span
                    class="ml-3 block font-normal truncate"
                  >
                    
            test
          
                  </span>
                </div>
                 
                <span
                  class="absolute inset-y-0 right-0 flex items-center pr-4"
                >
                  <svg
                    class="icon-test stroke-current text-danger w-8"
                    fill="currentColor"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </li>
              <li
                class="text-gray-900 cursor-default select-none relative py-2 pl-3 pr-9 hover:bg-secondary hover:text-white"
                id="listbox-item-1"
                role="option"
              >
                <div
                  class="flex items-center"
                >
                  <span
                    class="ml-3 block font-normal truncate"
                  >
                    
            test2
          
                  </span>
                </div>
                 
                <!---->
              </li>
            </ul>
          </div>
        </div>
      </div>
       
      <button
        class="group transition duration-150 ease-in-out relative w-full mt-5 shadow-sm flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-secondary lg:hover:bg-secondary-hovered lg:active:bg-secondary-hovered focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 ac"
        type="submit"
      >
        <span
          class="absolute left-0 inset-y-0 flex items-center pl-3"
        >
          <svg
            aria-hidden="true"
            class="h-5 w-5 text-white group-hover:text-white"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5 9V7a5 5 0 0110 0v2a2 2 0 012 2v5a2 2 0 01-2 2H5a2 2 0 01-2-2v-5a2 2 0 012-2zm8-2v2H7V7a3 3 0 016 0z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        
    Find Response
  
      </button>
    </div>
     
    <div
      class="rounded-md w-full md:w-1/2 my-2 border-secondary border"
    >
      <div
        class="clearfix flex items-end "
      >
        <img
          alt=""
          class="inline-block h-12 w-12 md:h-16 md:w-16 rounded-full ring-2 ring-white"
          src="https://images.unsplash.com/photo-1550525811-e5869dd03032?ixlib=rb-1.2.1&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
        />
         
        <div
          class="bg-secondary w-1/2 mx-4 mt-2 mb-8 p-2 rounded-lg text-white"
        >
          
    Hi can you tell me why you are not at work ?
  
        </div>
      </div>
       
      <div
        class="clearfix flex items-end flex-row-reverse"
      >
        <img
          alt=""
          class="inline-block h-12 w-12 md:h-16 md:w-16 rounded-full ring-2 ring-white"
          src="https://images.unsplash.com/photo-1550525811-e5869dd03032?ixlib=rb-1.2.1&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
        />
         
        <div
          class="bg-danger w-1/2 mx-4 mt-2 mb-8 p-2 rounded-lg text-white"
        >
          
    I dont know
  
        </div>
      </div>
    </div>
  </div>
</div>
`;
